### PR TITLE
Add translations for Option Type, Option Value, Property, & Product Property

### PR DESCRIPTION
**Pull request by nciemniak**
Builds upon `feature/mobility_gem_base` branch. 

The `TranslatableResourceScopes` Concern was created to easily perform joins on translation tables. By default, if you join on a translatable resource, it does not automatically join on its translation table, which means that filtering by a translatable field will not work. This module can be added to any resource that needs to perform a join on a translatable - for example, if the Product resource needs to join on Option Types and filter on an option, you can include the `TranslatableResourceScopes` module to facilitate the join filtering.

Translation tables created for OptionType, OptionValue, Property, and ProductProperty resources. Migrations were also created to transfer data for each resource to its respective translation table. 

### Option Type
* name
* presentation

### Option Value
* name
* presentation

### Property
* name
* presentation
* filter_param

### Product Property
* value
* filter_param

### CODE REVIEW

File: core/app/models/concerns/spree/translatable_resource_scopes.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Spree
+  module TranslatableResourceScopes
+    extend ActiveSupport::Concern
+
+    class_methods do
+      # To be used when joining on the resource itself does not automatically join on its translations table
+      # This method is to be used when you've already joined on the translatable table itself
+      def join_translation_table(translatable_class)
```
Comment (rafalcymerys) [inline core/app/models/concerns/spree/translatable_resource_scopes.rb:8]: Yes, that's a great addition

**Review by rafalcymerys — APPROVED** [review]

**Comment by rafalcymerys** [comment]
@nciemniak love it! I saw that some specs failed but it seems unrelated. I reran them - once they pass please merge

File: core/app/models/concerns/spree/translatable_resource_scopes.rb (hunk 1 above)
Comment (nciemniak) [reply core/app/models/concerns/spree/translatable_resource_scopes.rb:8]: 💪💪💪

**Comment by nciemniak** [comment]
@rafalcymerys  yeah, looks like there are still specs that randomly fail, but I don't think it has to do with the locale this time. Agree, doesn't seem related to these changes.

**Pull request merged**